src/Configuration/CFGCommon/CFGCommon.cpp

```diff
@@ -313,3 +316,61 @@ int CFG_find_u32_in_vector(const std::vector<uint32_t>& vector,
                            const uint32_t element) {
   return CFG_find_element_in_vector(vector, element);
 }
+
+int CFG_compiler_execute_cmd(const std::string& command,
+                             const std::string logFile, bool appendLog) {
+  if (m_execute_cmd_function != nullptr) {
+    return m_execute_cmd_function(command, logFile, appendLog);
+  } else {
+    std::string output = "";
+    return CFG_execute_cmd(command, output);
+  }
+}
+
+int CFG_execute_cmd(const std::string& cmd, std::string& output) {
+  int exitcode = 255;
+  char buffer[513];
+  std::string temp_string = "";
+  output = "";
+#if (defined(_MSC_VER) || defined(__MINGW32__))
+#define popen _popen
+#define pclose _pclose
+#define WEXITSTATUS
+#endif
+  CFG_POST_MSG("Command: %s", cmd.c_str());
+  std::string command = CFG_print("%s 2>&1", cmd.c_str());
+  FILE* pipe = popen(command.c_str(), "r");
+  if (pipe == nullptr) {
+    throw std::runtime_error("popen() failed!");
```

Comment:
This method will not be call from Raptor/Foedag. It is intended for the standalone programmer command line tool in [src/Configuration/Programmer/Programmer_cmd.cpp](https://github.com/os-fpga/FOEDAG/pull/1049/files#diff-a54213f6f00cb0e5fda51223209ca365e34409b16718112fa25902f269570de5). The implementation is pending for now. 

I think I should write some inline comments there to avoid confusion.

Reply:
Yes, please add those in next PR. It will be helpful for new users.